### DISCUSSION TITLE
Column sorting is not working with numbers(formatted like 1,234,45.00)

### DISCUSSION THREAD

**Discussion by rajesh-rk1** [primefaces/community#2868] [PrimeNG]
<img width="69" alt="image" src="https://github.com/user-attachments/assets/e5e25f30-0378-4f8e-804f-594e4ac51efe">

**Comment by qwadrox** [comment]
@rajesh-rk1 Hi!

It should work. This is what's happening internally: https://github.com/primefaces/primeng/blob/d6c7ffcfc5cb9a8ed355399ddae276c279836ccf/src/app/components/table/table.ts#L1593-L1605

Which PrimeNg version are you using? 
Could you create a sample on Stackblitz? 

```js
const a = '2,234,45.00';
const b = '1,234,45.00';

console.log(b.localeCompare(a));
console.log(a < b);

> -1
> false
```

**Reply by rajesh-rk1** [reply]
@qwadrox I am using  "primeng": "~17.8.0".
